[REG-1968] Docs for CVImage and CVText (#278)
Regression-Games/RGUnityBots: user-tools/README-CVText.md

- `data` - The data json object that defines how to evaluate this `CVText` criteria.
  - `text` - The text to find.  Note that this algorithm will treat this as a set of words to find, NOT a sentence.  Thus if you have 'Create New Profile' on a single line on one button, but have 'Create' 'New' 'Profile' each on a separate line in different screen area, the algorithm has to choose which one to select.  The current method is to select the smallest bounding area containing all required words that is also `withinRect` (if specified).
  - `textMatchingRule` - One of `Matches` or `Contains`
    - `Matches` - Each word in the provided `text` must be found as a whole word in the results.  This is a very exact matching rule and can sometimes lead to inconsistent results in frames with poor in game lighting on the text or low text contrast.

>     - `Contains` - Each word in the provided `text` must be found as a part of a word in the results.  This is a looser matching rule and if often used instead of `Matches` for more stable results.  For example "Time" would match to "Timer", "Time", "Times", "Timed", "Time", etc in the frame.  This may not seem ideal, but in most game situations, the contrast or text layout isn't always clearly identifiable and extra or incorrect letters may consistently be found for the text you are looking for in a specific frame of the game.  When you encounter those situations, you can adjust your game to give better text clarity/contrast to your users and/or you can utilize `Contains` instead of `Matches`.

Reviewer: ```suggestion
    - `Contains` - Each word in the provided `text` must be found as a part of a word in the results.  This is a looser matching rule and if often used instead of `Matches` for more stable results.  For example "Time" would match to "Timer", "Time", "Times", "Timed", "Time", etc in the frame.  This may not seem ideal but, in most game situations, the contrast or text layout isn't always clearly identifiable and extra or incorrect letters may consistently be found for the text you are looking for in a specific frame of the game.  When you encounter those situations, you can adjust your game to give better text clarity/contrast to your users and/or you can utilize `Contains` instead of `Matches`.
```

>     - `Contains` - Each word in the provided `text` must be found as a part of a word in the results.  This is a looser matching rule and if often used instead of `Matches` for more stable results.  For example "Time" would match to "Timer", "Time", "Times", "Timed", "Time", etc in the frame.  This may not seem ideal but, in most game situations, the contrast or text layout isn't always clearly identifiable and extra or incorrect letters may consistently be found for the text you are looking for in a specific frame of the game.  When you encounter those situations, you can adjust your game to give better text clarity/contrast to your users and/or you can utilize `Contains` instead of `Matches`.

Reviewer 2: ```suggestion
    - `Contains` - Each word in the provided `text` must be found as a part of a word in the results.  This is a looser matching rule and is often used instead of `Matches` for more stable results.  For example "Time" would match to "Timer", "Time", "Times", "Timed", "Time", etc in the frame.  This may not seem ideal but, in most game situations, the contrast or text layout isn't always clearly identifiable and extra or incorrect letters may consistently be found for the text you are looking for in a specific frame of the game.  When you encounter those situations, you can adjust your game to give better text clarity/contrast to your users and/or you can utilize `Contains` instead of `Matches`.
```

                {"leftButton":false,"middleButton":false,"rightButton":false,"forwardButton":false,"backButton":false,"scroll":{"x":0.0,"y":0.0},"duration":2.0 },
                {"leftButton":true,"middleButton":false,"rightButton":false,"forwardButton":false,"backButton":false,"scroll":{"x":0.0,"y":0.0} },
                {"leftButton":false,"middleButton":false,"rightButton":false,"forwardButton":false,"backButton":false,"scroll":{"x":0.0,"y":0.0} }
            ]
        }
    }
}
```

## Demos
**TODO:** Link to a public git repo files that are using this feature. **Feel free to skip this as I might be making a lot of these myself.**
- `endCriteria` - (Non-Public) https://github.com/Regression-Games/RGBossRoom/tree/main/Assets/RegressionGames/Resources/BotSegments/computer_vision/text_discover_criteria_menu_changeprofilebutton
- `botAction` - (Non-Public) https://github.com/Regression-Games/RGBossRoom/tree/main/Assets/RegressionGames/Resources/BotSegments/computer_vision/text_discover_action_menu_changeprofilebutton

## Limitations and Notes
The `CVText` type is still in an experimental phase and may provide inconsistent or unexpected results in some situations.

> - `textCaseRule` must always be set to `Ignore`.  The AIService does not currently consider capitalization in its results.

Reviewer 3: The user doesn't know about AIService so I'd say that `textCaseRule` is just currently unsupported

Author: i'll update this line accordingly

Author: Actually, I think I'll leave it, AIService as a concept is discussed throughout this doc as an external service.

- Text with low contrast relative to its background may not be detected or may detect incorrect characters
- Very small or very large text may not be detected or may detect incorrect characters

> - Some less traditional fonts may not be detected or may detect incorrect characters

Reviewer: ```suggestion
- Less common fonts may not be detected or may detect incorrect characters
```
